crates/txpool/src/ready.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Transaction readiness.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Readiness {
+    /// The transaction is stale (and should/will be removed from the pool).
+    Stale,
+    /// The transaction is ready to be included in pending set.
+    Ready,
+    /// The transaction is not yet ready.
+    Future,
+}
+
+/// A readiness indicator.
+pub trait Ready<T> {
```

Comment:
Suggestion: this trait in some way resembles the [Future](https://doc.rust-lang.org/std/future/trait.Future.html) trait. Perhaps we can use that interface to allow nicer code syntax: `let read_tx = tx.await?;`?